File: go/fiber/internal/modules/auth/service.go
package auth

import (
	"tcc/internal/common/enum"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/auth/dto"
	auth "tcc/internal/modules/auth/interfaces"
	"tcc/internal/modules/client"
	"tcc/internal/modules/manager"
	"time"
)

type AuthService struct{
	repository struct{
		client client.ClientRepository
		manager manager.ManagerRepository
	}
}

func (s *AuthService) LoginClient(data dto.LoginAuthDto) (*dto.LoginAuthClientOutputDto, *errors.BaseError) {
	client, _ := s.repository.client.FindByEmail(data.Email)
	if client == nil {
		return nil, errors.BadRequestError(messages.ErrorMessages.Auth.BadRequest)
	}

	if !(helper.IsPasswordCorrect(data.Senha, client.Senha)) {
		return nil, errors.BadRequestError(messages.ErrorMessages.Auth.BadRequest)
	}

	token, _ := helper.GenerateJWT(auth.JwtPayload{
		ID: client.ID.String(),
		Email: client.Email,
		CPF: client.CPF,
		Role: enum.RoleEnum(enum.CLIENT),	
	})

	return &dto.LoginAuthClientOutputDto{
		Token: token.Token,
		Usuario: *client,
		ExpiraEm: token.ExpiresIn.Format(time.RFC3339),
	}, nil
}

func (s *AuthService) LoginManager(data dto.LoginAuthDto) (*dto.LoginAuthManagerOutputDto, *errors.BaseError) {
	manager, _ := s.repository.manager.FindByEmail(data.Email)
	if manager == nil {
		return nil, errors.BadRequestError(messages.ErrorMessages.Auth.BadRequest)
	}

	if !(helper.IsPasswordCorrect(data.Senha, manager.Senha)) {
		return nil, errors.BadRequestError(messages.ErrorMessages.Auth.BadRequest)
	}

	token, _ := helper.GenerateJWT(auth.JwtPayload{
		ID: manager.ID.String(),
		Email: manager.Email,
		CPF: manager.CPF,
		Role: enum.RoleEnum(enum.MANAGER),	
	})

	return &dto.LoginAuthManagerOutputDto{
		Token: token.Token,
		Usuario: *manager,
		ExpiraEm: token.ExpiresIn.Format(time.RFC3339),
	}, nil
}

File: rust/axum/src/modules/auth/service.rs
use crate::modules::client::repository::ClientRepository;
use crate::modules::manager::repository::ManagerRepository;
use crate::common::errors::base::BaseError;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::errors::internal_server::internal_server_error;
use crate::common::messages::error::ErrorMessages;
use crate::common::enums::role::RoleEnum;
use crate::common::helper::jwt::generate_jwt;
use crate::common::helper::bcrypt::is_password_correct;
use crate::modules::auth::interfaces::JwtPayload;
use super::dto::{LoginAuthDto, LoginAuthClientOutputDto, LoginAuthManagerOutputDto};
use chrono::{DateTime, Utc};

pub struct AuthService {
}

impl AuthService {
    pub async fn login_client(
        data: LoginAuthDto,
    ) -> Result<LoginAuthClientOutputDto, BaseError> {

        
        let client = ClientRepository::find_by_email(&data.email).await
        .unwrap()
        .ok_or_else(|| bad_request_error(ErrorMessages::AUTH_BAD_REQUEST, None))?;
        

        if !is_password_correct(&data.senha, &client.senha) {
            return Err(bad_request_error(ErrorMessages::AUTH_BAD_REQUEST, None));
        }

        let payload = JwtPayload {
            id: client.id.to_string(),
            email: client.email.clone(),
            cpf: client.cpf.clone(),
            role: RoleEnum::Client,
        };

        let token = generate_jwt(payload)
            .map_err(|e| internal_server_error(ErrorMessages::INTERNAL_SERVER, None))?;

        let datetime: DateTime<Utc> = token.expires_in.into();
        Ok(LoginAuthClientOutputDto {
            token: token.token,
            usuario: client,
            expira_em: datetime.to_rfc3339(),
        })
    }

    pub async fn login_manager(
        data: LoginAuthDto,
    ) -> Result<LoginAuthManagerOutputDto, BaseError> {
        let manager = ManagerRepository::find_by_email(&data.email).await
        .unwrap()
        .ok_or_else(|| bad_request_error(ErrorMessages::AUTH_BAD_REQUEST, None))?;

        if !is_password_correct(&data.senha, &manager.senha) {
            return Err(bad_request_error(ErrorMessages::AUTH_BAD_REQUEST, None));
        }

        let token = generate_jwt(JwtPayload {
            id: manager.id.to_string(),
            email: manager.email.clone(),
            cpf: manager.cpf.clone(),
            role: RoleEnum::Manager,
        }).map_err(|e| internal_server_error(ErrorMessages::INTERNAL_SERVER, None))?;

        let datetime: DateTime<Utc> = token.expires_in.into();
        Ok(LoginAuthManagerOutputDto {
            token: token.token,
            usuario: manager,
            expira_em: datetime.to_rfc3339(),
        })
    }
} 